### PR TITLE
Replace MD4 with org.bountycastle library

### PR DESCRIPTION
**Pull request by Beacze**

### CODE REVIEW

File: Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java [hunk 1]
```diff
@@ -13,19 +13,18 @@
...
 
-import edu.sysu.virgoftp.ftp.encrypt.MD4;
+import org.junit.Test;
 
 public class MD4Test {
     static HashMap<String, byte[]> testData;
 
     static {
-        testData = new HashMap<String, byte[]>();
+        testData = new HashMap<>();
```
Comment (stweil) [inline Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java]: What is the reason for this change?

File: Kitodo/src/main/java/de/sub/goobi/helper/ldap/LdapUser.java [hunk 2]
```diff
@@ -77,6 +76,7 @@ public LdapUser() {
      */
     public void configure(User inUser, String inPassword, String inUidNumber)
             throws NamingException, NoSuchAlgorithmException, IOException, InterruptedException {
+        JDKMessageDigest.MD4 digester = new JDKMessageDigest.MD4();
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/helper/ldap/LdapUser.java]: `JDKMessageDigest` could be removed or did I miss something why it should used in this way?

File: Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java [hunk 3]
```diff
@@ -38,16 +37,12 @@
     }
 
     @Test
-    public void encryptTest() {
+    public void encryptTest() throws Exception {
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java]: Why did you not use more specific `UnsupportedEncodingException` instead of the really general `Exception`?

Comment (Beacze) [inline Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java]: I think for tests using more specific exception is not necessary. In this way method is shorter.

File: Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java (hunk 1 above)
Comment (Beacze) [inline Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java]: It is not necessary to write type of key and value in this HashMap. It is already given in line 24.

File: Kitodo/src/main/java/de/sub/goobi/helper/ldap/LdapUser.java (hunk 2 above)
Comment (Beacze) [inline Kitodo/src/main/java/de/sub/goobi/helper/ldap/LdapUser.java]: Fixed

File: Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java (hunk 3 above)
Comment (henning-gerhardt) [inline Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java]: It should be OK for tests, but not somewhere else.

Comment (stweil) [inline Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java]: ... as long as static code analyzers don't complain.

Comment (Beacze) [inline Kitodo/src/test/java/de/sub/goobi/helper/encryption/MD4Test.java]: It doesn't compalin
